pkg/pgmodel/migrate.go

```diff
@@ -58,101 +71,251 @@ func (t *mySrc) replaceSchemaNames(r io.ReadCloser) (io.ReadCloser, error) {
+
 	if err != nil {
-		return
+		return version, err
+	}
+
+	for res.Next() {
+		err = res.Scan(&version)
 	}
-	r, err = t.replaceSchemaNames(r)
-	return
+
+	if err != nil {
+		return version, err
+	}
+
+	return version, nil
 }
 
-func metadataUpdate(db *sql.DB, withExtension bool, key string, value string) {
-	/* Ignore error if it doesn't work */
-	if withExtension {
-		_, _ = db.Exec(metadataUpdateWithExtension, key, value, true)
-	} else {
-		_, _ = db.Exec(metadataUpdateNoExtension, key, value, true)
+func setDBVersion(tx pgx.Tx, version *semver.Version) error {
+	if _, err := tx.Exec(context.Background(), fmt.Sprintf(truncateMigrationsTable, migrationsTableName)); err != nil {
+		return fmt.Errorf("unable to truncate migrations table: %w", err)
 	}
+
+	if _, err := tx.Exec(context.Background(), fmt.Sprintf(setVersion, migrationsTableName), version.String()); err != nil {
+		return fmt.Errorf("unable to set version in DB: %w", err)
+	}
+
+	return nil
 }
 
 var migrateMutex = &sync.Mutex{}
 
-// Migrate performs a database migration to the latest version
-func Migrate(db *sql.DB, versionInfo VersionInfo) (err error) {
-	migrateMutex.Lock()
-	defer migrateMutex.Unlock()
-	// The migration table will be put in the public schema not in any of our schema because we never want to drop it and
-	// our scripts and our last down script drops our shemas
-	driver, err := postgres.WithInstance(db, &postgres.Config{MigrationsTable: "prom_schema_migrations"})
+func execMigrationFiles(tx pgx.Tx, name string) error {
+	f, err := migrations.MigrationFiles.Open(name)
 	if err != nil {
-		return fmt.Errorf("cannot create driver due to %w", err)
+		return fmt.Errorf("unable to get migration scripts: name %s, err %w", name, err)
 	}
 
-	_, err = db.Exec(timescaleInstall)
-	if err != nil {
-		return fmt.Errorf("timescaledb failed to install due to %w", err)
+	var (
+		entries []os.FileInfo
+		entry   os.FileInfo
+		file    http.File
+	)
+
+	if myToC, ok := toc[name]; ok {
+		entries = make([]os.FileInfo, 0, len(myToC))
+		for _, fileName := range myToC {
+			fullName := filepath.Join(name, fileName)
+			file, err = migrations.MigrationFiles.Open(fullName)
+			if err != nil {
+				return fmt.Errorf("unable to get migration script from toc: name %s, err %w", fullName, err)
+			}
+
+			if entry, err = file.Stat(); err != nil {
+				return fmt.Errorf("unable to stat migration script from toc: name %s, err %w", fullName, err)
+			}
+
+			entries = append(entries, entry)
+		}
+	} else {
+		entries, err = f.Readdir(-1)
+		if err != nil {
+			return fmt.Errorf("unable to read migration scripts directory: name %s, err %w", name, err)
+		}
+
+		sort.Slice(entries, func(i, j int) bool {
+			first, err := strconv.Atoi(strings.Split(entries[i].Name(), "-")[0])
+			if err != nil {
+				return false
+			}
+			second, err := strconv.Atoi(strings.Split(entries[j].Name(), "-")[0])
+			if err != nil {
+				return false
+			}
+
+			return first < second
```

Comment:
Is this checking file-order based on the assumption that the filenames are are `<filename>-<order num>`?
I'm betting this would read better split out into it's own function.

I'm also  not sure of the behavior that files without a version are incomparable, that might break things.

(Also, it probably doesn't matter, but it bugs me a little that we split and parse the string every time we call the ordering function, instead of just doing so once at the start)